twitterMini/Twitter.swift

```diff
@@ -0,0 +1,130 @@
+    let createdAt: String
+    let text: String
+    let user: TwitterUser
+    let likes: String
+    let retweets: String
+    //TODO: deal with media (["extendedEntities"]["media"]["media_url_https"])
+}
+
+struct TwitterUser {
+    let screenName: String
+    let name: String
+    let profileImageURL: String
+}
+
+// MARK: API Client
+
+class TwitterAPIClient {
+    static let sharedClient = TwitterAPIClient()
+    
+    let twitterBaseURL = "https://api.twitter.com/1.1/"
+    let callback = "twittestIntrepid://"
+    
+    let oauthManager = OAuth1Swift(
+        consumerKey:    "CbE3BbqhqwxNbRdGCP5y2gDU2",
+        consumerSecret: "PxQy9075tg4QSZsLZNX3wwXEkG5fyuhtrAlNjM868sYRi6BCU8",
+        requestTokenUrl: "https://api.twitter.com/oauth/request_token",
+        authorizeUrl:    "https://api.twitter.com/oauth/authorize",
+        accessTokenUrl:  "https://api.twitter.com/oauth/access_token"
+    )
+    private var idForLoggedInUser: String?
+    
+    func authorizeTwitter(completion: (success: Bool) -> Void){
+        if let callbackURL = NSURL(string: callback) {
+            oauthManager.authorizeWithCallbackURL( callbackURL, success: {
+                credential, response, parameters in
+                print("Authentication Success, userID:\(parameters["user_id"])")
+                self.idForLoggedInUser = parameters["user_id"]
+                completion(success:true)
+                }, failure: { error in
+                    print("Authentication Error: \(error.localizedDescription)")
+                    completion(success:false)
+                }
+            )
+        }
+    }
+
+    func fetchTimeline(completion: (rawStatuses: [[String:AnyObject]]?, error: NSError?) -> Void){
+        if let idForLoggedInUser = idForLoggedInUser{
+            let homeTimelineEndpoint = "statuses/home_timeline.json"
+            let homeTimelineURL = "\(twitterBaseURL)\(homeTimelineEndpoint)?user_id=\(idForLoggedInUser)"
+            print("hitting \(homeTimelineURL)")
+            oauthManager.client.get(homeTimelineURL, success: { data, response in
+                print("Fetch Home Timeline Success")
+                do {
+                    let dataObject = try NSJSONSerialization.JSONObjectWithData(data, options: []) as? [[String:AnyObject]]
+                    completion(rawStatuses: dataObject, error: nil)
+                } catch {
+                    print("Error Serializing JSON: \(error)")
+                    completion(rawStatuses: nil, error: error as NSError)
+                }
+                }, failure: { error in
+                    print("Fetch Home Timeline Error:\(error.localizedDescription)")
+                    completion(rawStatuses: nil, error: error)
+            })
+        } else {
+            print("user is not logged in...")
+            // FIXME: use constants
+            let error = NSError(domain: "Tom's Domain", code: 420, userInfo: [NSLocalizedDescriptionKey:"There is no authenticated user."])
```

Comment:
You should centralize these error constants in a single place, such as NSError+TwitterMini.swift and give it a helper constructor like
`NSError.createErrorWithErrorType(TwiterMiniError.NotAuthenticated)`